### PR TITLE
Solution: SPRINT CHALLENGE -- Lambda Times React

### PR DESCRIPTION
**Pull request by nickcannariato**
## Minimum Viable Product

- [x] Go through the `Tabs`, `Tab`, `Cards`, and `Card` components following the instructions, and passing data and props to get the tabs and cards to appear on the screen.
- [x] Once the Tabs and Cards are rendering to the screen complete the `changeSelected` and `filterCards` functions in the `Content` component.
- [x] You should now be able to filter cards using your tabs!
- [x] Make sure all of your props being passed are validated using PropTypes.
- [x] Find the `TopBar` and `Header` components. Convert these two components to Styled Components. You should not have any `className` props when you are finished.

## Stretch Challenge

There are multiple stretch challenges available to you, you may attempt these in any order. Remember, stretch challenges are only to give you extra time to work on these concepts, if you do not get to these challenges, that is fine! Continue working on your main objectives.

- [ ] Re-factor the app, so that it uses ALL styled components. There should be no `className` props on any component. To truly test this, delete the CSS file.

- [ ] You will find a `Carousel` component in your Content folder. Complete this component, rendering a functional carousel. Add this component between your `Tabs` and `Cards` components within the `Content` component. Added challenge: make it so that there is infinite scroll to the right and the left.

- [ ] Add a login and an HOC. Make it so that when a user clicks on the login button at the top, a login modal is shown (Use React-strap). Have a user login, validating the login credentials on the `localStorage`. Add a Higher Order Component that wraps the `Content` component, only allowing it to render once a user has logged in. For more instructions see this README: [React-Insta-Clone: Day III](https://github.com/LambdaSchool/React-Insta-Clone/blob/master/DAY_THREE_README.md#tasks-day-iii)

### CODE REVIEW

**Comment by nickcannariato** [comment]
@sabrinagear MVP has now been met. I'm going to quickly knock out the styled-components stretch goal and then deploy it to either Netlify or Zeit. Will drop a link here when the deployment is ready.

**Comment by nickcannariato** [comment]
App is deployed here: https://birdcar-lambda-times.netlify.com/

File: ANSWERS.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Answers
+
+## What are PropTypes used for? Please describe why it's important to type check our data in JavaScript.
+
+> PropTypes is an object that allows you to define what props your component requires and what type they should be. This is vitally important in a real world application because it allows you to ensure that the data being passed into your components is either expected or explicitly handled in an error. It also means that future developers on your team can reason about your application better since they now have a well defined data flow.
+
```
Comment (sabrinagear) [inline ANSWERS.md]: Yes, excellent analysis.

File: ANSWERS.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+## Describe a life-cycle event in React?
+
+> A lifecycle event is an event, defined by the React library, that occurs throughout the various "life stages" of a component. These life stages are defined as mounting, updating, and unmounting, and are analagous to birth, life, and death. A specific example of a lifecycle event would be `componentDidMount()` which occurs at the end of the mounting phase and executes as the updating phase begins.
+
+## Explain the details of a Higher Order Component?
+
+> A Higher Order Component, much like a higher order function, is a function that accepts a component and returns a new component that renders the component passed into it with enhanced functionality.
+
```
Comment (sabrinagear) [inline ANSWERS.md]: Yes, I would also note that a higher order component is, in fact, a higher function.

File: ANSWERS.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+## What are three different ways to style components in React? Explain some of the benefits of each.
+
+> You can style React components in at least 4 different ways that I can think of, but an example of 3 ways to style React compoennts would be: Inline styles, CSS stylesheets, & styled components. 
+> 
+> The benefit of Inline styles is that if you just need to adjust a couple of styles on the component in a specific context (adding just a touch more padding in location, for example) then you can pass an object to **just** that instance of the component. 
+>
+> The benefit of using CSS stylesheets is that it separates your design and functionality concerns into separate files and provides you a reusable series of CSS styles that you can pass as `className`'s to various components to bring about the desired styles
+>
+> Finally, Styled Components provides us with the benefit of defining styles inside of the same file as our react component, which provides us with a "single source of truth" for the component's look and feel and also allows us to have styles that are scoped to **only** apply to that component.
```
Comment (sabrinagear) [inline ANSWERS.md]: Yes, well reasoned.

**Pull request merged**